treescript/src/treescript/merges.py

```diff
@@ -115,6 +115,28 @@ def touch_clobber_file(repo_path):
             "Merge old head via |hg debugsetparents {} {}| CLOSED TREE DONTBUILD a=release".format(tagged_from_rev, to_branch),
             repo_path=repo_path,
         )
+        # Preserve tags.
+        patch_file = os.path.join(os.path.dirname(repo_path), "patch_file")
+        tag_diff = await run_hg_command(config, "diff", "-r", to_branch, os.path.join(repo_path, ".hgtags"), "-U9", return_output=True, repo_path=repo_path)
+        with open(patch_file, "w") as fh:
+            fh.write(tag_diff)
+        await run_command(["patch", "-R", "-p1", "-i", patch_file], cwd=repo_path)
+        os.unlink(patch_file)
+        with open(os.path.join(repo_path, ".hgtags"), "a") as fh:
+            # Skip four header lines
+            for line in tag_diff.splitlines()[4:]:
+                # We only care about additions.
+                if not line.startswith("+"):
+                    continue
+                line = line.replace("+", "")
```

Comment:
We should strip the first character, rather than replace, in case there are tags with `+` in them.